src/proxy_configuration.js

```diff
@@ -51,6 +52,7 @@ const CHECK_ACCESS_MAX_ATTEMPTS = 2;
  *   Custom function that allows you to generate the new proxy URL dynamically. It gets the `sessionId` as a parameter
  *   and should always return stringified proxy URL.
  *   This function is used to generate the URL when {@link ProxyConfiguration.newUrl} or {@link ProxyConfiguration.newProxyInfo} is called.
+ * @property {ProxyConfigurationFunction} [useApifyProxy]
```

Comment:
What's this?

Reply:
It was in the code before (apparently added by you btw), I just added it to the jsdoc to have better type safety, as it was missing there, but as the comment says, it is some legacy flag.

> // This is not an actual param, but it's here for legacy purposes.
> useApifyProxy: ow.optional.boolean,

Obviously it should be a boolean, not `ProxyConfigurationFunction`, that's a typo. Or should I remove it as a whole? It is also noted in the docs of `createProxyConfiguration()`, so even if it is a legacy option, it is still present in the docs, so why not have it in the options type?

Reply:
It's only there to allow configuration with Input UI from the platform. I left it out from the docs on purpose I guess, because we don't want anyone to use it. But at the same time, it needs to be in the `ow` check because its an `exactShape` check.

We use `exactShape` checks because people were often passing constructor options to bad places or with typos and without the `exactShape` it silently continues, but it does not work as expected.

Reply:
Looking at the code, I think the best way to deal with this would be to copy the options in the `createProxyConfiguration` function and delete the `useApifyProxy` property from the opts that are passed down to `ProxyConfiguration`. The option is not relevant to `ProxyConfiguration` at all.

Reply:
Actually there is a test that uses `useApifyProxy` in the ctor, so we need to keep in the in the `ow` check (or adjust that test). 

https://github.com/apify/apify-js/blob/master/test/proxy_configuration.test.js#L70